src/providers/sharedData/SharedDataProvider.ts

```diff
@@ -0,0 +1,78 @@
+import {ICredential} from "@sphereon/ssi-types";
+import {ProofType, UnsignedCredential, VerifiableCredential} from "@veramo/core";
+import * as RNFS from 'react-native-fs';
+import ShareMenuModule from 'react-native-share-menu';
+
+import {NavigationBarRoutesEnum, ScreenRoutesEnum} from "../../@types";
+import {SharedItem} from "../../@types/files";
+import {translate} from "../../localization/Localization";
+import * as RootNavigation from '../../navigation/rootNavigation'
+import store from "../../store";
+import {storeVerifiableCredential} from "../../store/actions/credential.actions";
+import {showToast, ToastTypeEnum} from "../../utils/ToastUtils";
+import {toCredentialSummary} from "../../utils/mappers/CredentialMapper";
+
+class SharedDataProvider {
+
+  public static async getDataOnStartup() {
+    await ShareMenuModule.getSharedText((data: any) => {
+      SharedDataProvider.receiveData(data);
+    });
+  }
+
+
+
+  private static onDecline() {
+    return async () => RootNavigation.navigate(NavigationBarRoutesEnum.HOME, {
+      screen: ScreenRoutesEnum.CREDENTIALS_OVERVIEW
+    });
+  }
+
+  private static onAcceptance(storeCredential: (vc: VerifiableCredential) => Promise<void>, vc: UnsignedCredential & { proof: ProofType }) {
```

Comment:
not to important, but call it onAccept as it is then inline with eveything else

Reply:
attempted a fix